### PR TITLE
DATAUP-330: util module test fixes

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

- Clean up linting issues in the utils modules
- Ensure that all tests in the `util` directory run expectations.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-330
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by running `make test-frontend-unit`

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/util/display.js [hunk 1]
```diff
@@ -83,22 +69,6 @@ define([
...
-     * caption centered below.
-     */
-    function loadingSpinner(caption) {
-        let spinner = '<span class="fa fa-spinner fa-pulse fa-2x fa-fw">';
-        if (caption) {
-            spinner += caption + '... &nbsp; &nbsp;';
-        }
-        spinner += '</span>';
-        return spinner;
-    }
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/display.js:100]: Unused anywhere. There is a very similar function in the `UI` module.

File: kbase-extension/static/kbase/js/util/timeFormat.js [hunk 2]
```diff
@@ -336,17 +336,6 @@ define([], () => {
...
 
-    /**
-     * @private
-     * @method
-     * Checks if the timestamp looks like it should be ISO 8601 format.
-     * If so, it "cleans" it up to make sure (if it has a trailing +0000, it converts it
-     * to a trailing +00:00, for example)
-     */
-    function cleanupISOFormat(timestamp) {
-        // check if ISO.
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/timeFormat.js:347]: not as useful a function as the name might suggest

File: kbase-extension/static/kbase/js/util/timeFormat.js [hunk 3]
```diff
@@ -263,8 +261,10 @@ define([], () => {
             seconds = 0;
         }
 
-        const exactDate = function (date) {
-            return monthLookup[date.getMonth()] + ' ' + date.getDate() + ', ' + date.getFullYear();
+        const exactDate = function (_date) {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/timeFormat.js:264]: `date` defined in the upper scope

File: kbase-extension/static/kbase/js/util/timeFormat.js [hunk 4]
```diff
@@ -81,18 +81,16 @@ define([], () => {
...
-                // This would mean its got the 'Invalid Date' status.
+                // This would mean it has the 'Invalid Date' status.
                 if (isNaN(d.getTime())) {
                     return null;
-                } else {
-                    d.setFullYear(t[0]);
-                    return d;
                 }
+                d.setFullYear(t[0]);
+                return d;
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/timeFormat.js:89]: remove unneeded `else`s

File: test/unit/spec/util/BootstrapDialogSpec.js [hunk 5]
```diff
@@ -31,12 +30,13 @@ define(['jquery', 'util/bootstrapDialog'], ($, Dialog) => {
             expect($($.find('.fade.in')).is(':visible')).toBeTruthy();
         });
 
-        it('Should hide on command', () => {
-            simpleDialog.show();
-            simpleDialog.hide();
+        it('Should hide on command', (done) => {
             simpleDialog.$modal.on('hidden.bs.modal', () => {
                 expect(simpleDialog.$modal.is(':visible')).toBeFalsy();
+                done();
```
Comment (ialarmedalien) [inline test/unit/spec/util/BootstrapDialogSpec.js:36]: make sure test runs and completes

File: test/unit/spec/util/BootstrapSearchSpec.js [hunk 6]
```diff
@@ -21,7 +22,8 @@ define(['jquery', 'util/bootstrapSearch', 'base/js/namespace'], ($, BootstrapSea
 
         it('Should fire an input function when triggered by input', (done) => {
             const bsSearch = new BootstrapSearch($targetElem, {
-                inputFunction: () => {
+                inputFunction: (event) => {
+                    expect(event.type).toBe('input');
```
Comment (ialarmedalien) [inline test/unit/spec/util/BootstrapSearchSpec.js:26]: add in expectation (the test did not check anything before)

File: test/unit/spec/util/BootstrapSearchSpec.js [hunk 7]
```diff
@@ -89,24 +91,25 @@ define(['jquery', 'util/bootstrapSearch', 'base/js/namespace'], ($, BootstrapSea
...
+                expect(event.type).toBe('focus');
                 done();
             });
             bsSearch.focus();
         });
 
         it('Should set placeholder text', () => {
             const placeholder = 'some text';
-            const bsSearch = new BootstrapSearch($targetElem, {
+            new BootstrapSearch($targetElem, {
```
Comment (ialarmedalien) [inline test/unit/spec/util/BootstrapSearchSpec.js:103]: remove unused vars

**Review by briehl — APPROVED** [review]
:+1:  all positive changes.

**Pull request merged**